client/src/main/java/io/pravega/client/stream/TimeWindow.java

```diff
@@ -14,18 +14,34 @@
 /**
  * Represents a time window for the events which are currently being read by a reader.
  * 
- * The upper time bound is a timestamp which is greater than or equal to any that were provided by
- * any writer via the {@link EventStreamWriter#noteTime(long)} API prior to the current location in
- * the stream.
- * 
- * Similarly the lower time bound is a timestamp which is less than or equal to the most recent
+ * The lower time bound is a timestamp which is less than or equal to the most recent
  * value provided via the {@link EventStreamWriter#noteTime(long)} API for by any writer using that
- * API at the current location in the stream.
+ * API at the current location in the stream. If the reader is near the beginning of the truncation 
```

Comment:
Even if the stream is truncated, the timewindow with a position before truncation point can still be present in the watermark-segment. 
The computation of watermark is done after a writer submits a position and truncation can happen after it but the watermarks are all preserved in the watermark segment for each stream.
So even though timewindows can have a lower bound on time and position, but a reader can only really read from the truncation point. 

So i am not sure if the description here correctly reflects that.

Reply:
That is true. Though I'm not sure it is relevant to an end user. They can only read from the truncation point forward and the time will reflect their position in the stream, so the fact that marks exist below this may or may not allow for more accurate time information if they are at the truncation point, but as this cannot be guaranteed, it is being disclaimed here.